### PR TITLE
Simplify connection state related enums

### PR DESCRIPTION
**Pull request by hiroshihorie**

### CODE REVIEW

File: Sources/LiveKit/Core/Engine.swift [hunk 1]
```diff
@@ -48,7 +43,9 @@ internal class Engine: MulticastDelegate<EngineDelegate> {
     internal struct State {
         var url: String?
         var token: String?
-        var connectionState: ConnectionState = .disconnected(reason: .sdk)
+        var isReconnect: Bool = false
+        var reconnectMode: ReconnectMode?
```
Comment (davidzhao) [inline Sources/LiveKit/Core/Engine.swift:47]: nit: we could get rid of isReconnect and use `reconnectMode = .none`

File: Sources/LiveKit/Types/ConnectionState.swift [hunk 2]
```diff
@@ -21,30 +21,30 @@ public enum ReconnectMode {
...
-
-    public func isEqual(to rhs: ConnectMode, includingAssociatedValues: Bool = true) -> Bool {
-        switch (self, rhs) {
-        case (.reconnect(let mode1), .reconnect(let mode2)): return includingAssociatedValues ? mode1 == mode2 : true
-        case (.normal, .normal): return true
-        default: return false
-        }
-    }
-}
+// public enum ConnectMode {
```
Comment (davidzhao) [inline Sources/LiveKit/Types/ConnectionState.swift:24]: remove?

File: Sources/LiveKit/Types/ConnectionState.swift [hunk 3]
```diff
@@ -56,21 +56,41 @@ extension ConnectionState: Identifiable {
...
+            return true
+        case (.connecting, .connecting):
+            return true
+        case (.connected, .connected):
+            return true
         default: return false
         }
     }
 
+    //    public func isEqual(to rhs: ConnectionState, includingAssociatedValues: Bool = true) -> Bool {
```
Comment (davidzhao) [inline Sources/LiveKit/Types/ConnectionState.swift:71]: remove?

File: Sources/LiveKit/Core/Engine.swift [hunk 4]
```diff
@@ -70,14 +66,15 @@ internal class Engine: MulticastDelegate<EngineDelegate> {
         ConnectivityListener.shared.add(delegate: self)
 
         // trigger events when state mutates
-        self.state.onMutate = { [weak self] oldState, newState in
+        self.state.onMutate = { [weak self] state, oldState in
```
Comment (davidzhao) [inline Sources/LiveKit/Core/Engine.swift:69]: ohh was the order the other way around?

**Review by davidzhao — APPROVED** [review]
looks good!

File: Sources/LiveKit/Core/Engine.swift (hunk 4 above)
Comment (hiroshihorie) [reply Sources/LiveKit/Core/Engine.swift:69]: I just changed the ordering in this PR 🙂

**Pull request merged**